searchcore/src/apps/proton/proton.cpp

```diff
@@ -186,8 +190,14 @@ App::Main()
         LOG(debug, "serviceidentity: '%s'", params.serviceidentity.c_str());
         LOG(debug, "subscribeTimeout: '%" PRIu64 "'", params.subscribeTimeout);
         std::chrono::milliseconds subscribeTimeout(params.subscribeTimeout);
-        config::ConfigUri identityUri(params.identity);
-        protonUP = std::make_unique<proton::Proton>(identityUri, _argc > 0 ? _argv[0] : "proton", subscribeTimeout);
+        FastOS_ThreadPool threadPool(128_Ki);
+        uint32_t numProcs = std::thread::hardware_concurrency();
+        FNET_Transport transport(fnet::TransportConfig(std::max(1u, std::min(4u, numProcs/8))));
```

Comment:
Consider adding a helper function for calculating TransportConfig.

Reply:
Xixed